protocol-tokio/src/protocol/codec/message.rs

```diff
@@ -88,17 +89,113 @@ pub enum Message {
     ProbeSocketAck(nt::LightWeightConnectionId),
     CreateNodeId(nt::LightWeightConnectionId, NodeId),
     AckNodeId(nt::LightWeightConnectionId, NodeId),
+    Bytes(nt::LightWeightConnectionId, Bytes),
+
+    UserMessage(M),
+}
 
-    GetBlockHeaders(nt::LightWeightConnectionId, GetBlockHeaders),
-    BlockHeaders(nt::LightWeightConnectionId, Response<BlockHeaders, String>),
-    GetBlocks(nt::LightWeightConnectionId, GetBlocks),
-    Block(nt::LightWeightConnectionId, Response<Block, String>),
-    SendTransaction(nt::LightWeightConnectionId, TxAux),
+pub enum ChainMessage<B, T>
```

Comment:
A comment explaining the purpose of `ChainMessage` would be nice.

Reply:
I'll reply here (as that reply goes to the other comments as well). The idea of the network protocol is that is has some `technical/control` messages and user messages go over the wire. The idea of the change was to keep 2 types one for the general in `protocol` library and one user specific. In our case `Message` itself is general and `ChainMessage` is application specific. The alternative is that user will always implement his `Message` that will keep control messages around. There are some conventions how to implement such a message so it can be put on the wire flat (i.e. not reuse the prefixes of the control one).
So this change does not extend network layout, but it allows to write the protocol in a generic way.